includes/namespace.php

```diff
@@ -29,63 +29,111 @@
 use WP_Error;
 use WP_Site;
 
-if (
-	! class_exists( '\Google\Web_Stories\Plugin' ) ||
-	! file_exists( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js' )
-) {
-	/**
-	 * Displays an admin notice about why the plugin is unable to load.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return void
-	 */
-	function _print_missing_build_admin_notice() {
-		?>
-		<div class="notice notice-error">
-			<p>
-				<strong><?php esc_html_e( 'Web Stories plugin could not be initialized.', 'web-stories' ); ?></strong>
-			</p>
-			<p>
-				<?php
-					echo wp_kses(
-						sprintf(
-						/* translators: %s: build commands. */
-							__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
-							'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
-						),
-						[
-							'code' => [],
-						]
-					);
-				?>
-			</p>
-		</div>
-		<?php
-	}
+// Load Compatibility class the old fashioned way.
+if ( ! class_exists( 'Compatibility' ) ) {
+	require_once WEBSTORIES_PLUGIN_DIR_PATH . 'includes/Compatibility.php';
+}
+
+global $web_stories_compatibility;
+
+$web_stories_error = new WP_Error();
+$extensions        = [
+	'date'   => [
+		'classes' => [
+			'DateTimeImmutable',
+		],
+	],
+	'dom'    => [
+		'classes' => [
+			'DOMAttr',
+			'DOMComment',
+			'DOMDocument',
+			'DOMElement',
+			'DOMNode',
+			'DOMNodeList',
+			'DOMText',
+			'DOMXPath',
+		],
+	],
+	'json'   => [
+		'functions' => [
+			'json_decode',
+			'json_encode',
+		],
+	],
+	'libxml' => [
+		'functions' => [
+			'libxml_use_internal_errors',
+		],
+	],
+	'spl'    => [
+		'functions' => [
+			'spl_autoload_register',
+		],
+	],
+];
+
+$web_stories_compatibility = new Compatibility( $web_stories_error );
+$web_stories_compatibility->set_extensions( $extensions );
+$web_stories_compatibility->set_php_version( WEBSTORIES_MINIMUM_PHP_VERSION );
+$web_stories_compatibility->set_wp_version( WEBSTORIES_MINIMUM_WP_VERSION );
+$web_stories_compatibility->set_js_path( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js' );
+$web_stories_compatibility->set_class_name( '\Google\Web_Stories\Plugin' );
+
+/**
+ * Displays an admin notice about why the plugin is unable to load.
+ *
+ * @since 1.0.0
+ *
+ * @return void
+ */
+function _print_missing_build_admin_notice() {
+	global $web_stories_compatibility;
+
+	$web_stories_compatibility->check_js_built();
+	$web_stories_compatibility->check_php_built();
+	$web_stories_compatibility->check_extensions();
+	$web_stories_compatibility->check_classes();
+	$web_stories_compatibility->check_functions();
```

Comment:
Should there be a single `check()` method that calls all of these?

Reply:
I want to wait for @swissspidy to chime in here, as I want to know what checks he wants where. But bundling checks together is simple enough to do.

Reply:
+1 to a single method

Reply:
I also agree, it was what I had planned in the IB. See https://github.com/google/web-stories-wp/issues/4791#issuecomment-705555631 . I called it `run_checks` instead of `checks`